### ISSUE TITLE
Add option to require the alt attribute on image fields.

### ISSUE THREAD

**Issue by gmclelland** [processwire/processwire-requests#11]
Can you add an option to image fields to require the alt attribute and have it enabled by default?  This would help make sites everywhere more accessible and help improve SEO.

Drupal 8 adopted this change here https://www.drupal.org/node/2303765

Here is what it looks like in Drupal 8:
![screen shot 2015-03-08 at 11 50 11 am](https://cloud.githubusercontent.com/assets/40570/18923329/6a11e74a-8571-11e6-832b-0c982113fb64.png)

Maybe the CKeditor wysiwyg editor's page image editor could also automatically prefill the alt attribute from the chosen image as default that can be optionally overridden?
![wysiwyg-alt-text](https://cloud.githubusercontent.com/assets/40570/18923535/353fa312-8572-11e6-8d41-7664b3afac80.jpg)

**Comment by adrianbj** [comment]
Somewhat related is this request: https://github.com/ryancramerdesign/ProcessWire/issues/1058 although in that case I think we were all talking about the description field attached to the image itself, rather than the one stored with the image in the alt tag.

**Comment by gRegorLove** [comment]
+1 for the editor always inserting an `alt` attribute, even if blank. I think CKEditor already does that, though.

I have mixed feelings on auto-populating the alt (with the filename?), or requiring it not be blank by default. Sometimes it makes a less than ideal experience for screenreaders, e.g. a photo+caption where the alt text duplicates information in the caption. The [guideline](https://indieweb.org/alt#Guidelines) I've been trying to follow is "Use alt text that makes sense when read out loud along with the context of any text before, and any text after the image." Previously I was just trying to describe the image and that gives bad results sometimes. "Photo of a waterfall" with caption "Waterfall" would be read "Photo of a waterfall waterfall." A filename as the alt would make it even worse.

**Comment by gmclelland** [comment]
Hrmmm... I'm not sure if there is confusion in this thread, so here is another image to try to clarify what I was trying to suggest.  This is really several issues here.

1. Add an option to make "Description (alt attribute)" on the image fields set to required
2. Make that new option required by default so all websites get the accessibility and SEO benefits.  This option could still be disabled per field if needed.
3. Change the ckeditor wysiwyg image editor to always show the Image description(alt attribute).  Right now it is hidden behind an icon.
4. If the ckeditor wysiwyg image editor's "Image description (alt attribute)" is empty, then prefill it with the "Description" from the image field's image.

Here is an image that hopefully illustrates the idea a little better
![imageeditor](https://cloud.githubusercontent.com/assets/40570/18933790/0c1645e8-859c-11e6-9819-e191dfbd1ae9.jpg)

**Comment by elabx** [comment]
Want to pop this up again since I think this is could exist just how @gmclelland proposes it.

**Comment by gmclelland** [comment]
Note: I also use a different CMS at work and they also include a checkbox for whether or not the image is decorative.

Under the checkbox is says the following:
[Decorative Images](https://www.w3.org/WAI/tutorials/images/decorative/) will use null alt text (alt=””). Only select for images that don’t add information to the content of a page and can be ignored by assistive technologies.

![Image](https://github.com/user-attachments/assets/a4bb5d10-c15c-4300-92a1-4c1770e66a77)

**Comment by elabx** [comment]
I'd think it should be enough to have another field aside from caption like in your screenshot. Why not just leave the alt text field empty? Is it to be super strict about the accessibility issue that will pop if leaving it empty?

**Comment by gmclelland** [comment]
Yes, it probably would be enough.  I just thought that was interesting way of handling decorative images.
